### PR TITLE
support multi-path gopath

### PR DESCRIPTION
**Pull request by wencan**
support multi-path gopath

### CODE REVIEW

**Comment by wencan** [comment]
@googlebot I signed it!

**Comment by codyoss** [comment]
@wencan can you please create an issue for this first to discuss. Thanks.

File: mockgen/parse.go [hunk 1]
```diff
@@ -641,13 +641,16 @@ func parsePackageImport(srcDir string) (string, error) {
...
-	goPath := os.Getenv("GOPATH")
-	if goPath == "" {
+	goPaths := os.Getenv("GOPATH")
+	if goPaths == "" {
 		return "", fmt.Errorf("GOPATH is not set")
 	}
-	sourceRoot := filepath.Join(goPath, "src") + string(os.PathSeparator)
-	if !strings.HasPrefix(srcDir, sourceRoot) {
-		return "", errOutsideGoPath
+	goPathList := strings.Split(goPaths, ":")
```
Comment (codyoss) [inline mockgen/parse.go:648]: This assumes UNIX.

> The GOPATH environment variable lists places to look for Go code.
On Unix, the value is a colon-separated string.
On Windows, the value is a semicolon-separated string.
On Plan 9, the value is a list.

**Comment by wencan** [comment]
@codyoss Thanks for your reminder.
I have changed it.

**Pull request merged**